src/hooks/useIntersectionObserver.ts

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useRef } from 'react';
+
+interface useIntersectionObserverProps {
+  callbackIn?: (() => void) | void;
+  callbackOut?: (() => void) | void;
+}
+
+// callbackIn: entry.isIntersecting === true 일 때 실행
+// callbackOut: entry.isIntersecting === false 일 때 실행
+export default function useIntersectionObserver({
+  callbackIn = () => {},
+  callbackOut = () => {},
+}: useIntersectionObserverProps) {
+  const observationTargetRef = useRef<HTMLElement | null>(null);
+
+  const observer = useRef(
+    new IntersectionObserver(([entry]) => {
+      // entry가 뷰포트 안에 있으면 callback 함수 호출
+      if (entry.isIntersecting) {
+        callbackIn();
```

Comment:
💬 
callback에 기본값으로 `() => {}` 를 주는 방법도 있지만, 기본값을 제거하고 함수에 선택적 실행을 적용해도 좋을 것 같습니다.


```suggestion
        callbackIn?.();
```